server/tournament/match_stats_min.py

```diff
@@ -0,0 +1,171 @@
+    match.stats.save()
+
+    return 200, match.stats
+
+
+def handle_score(
+    match_event: MatchEventCreateSchema, match: Match, team: Team
+) -> tuple[int, MatchStats | dict[str, str]]:
+    if match_event.scored_by_id is None or match_event.assisted_by_id is None:
+        return 422, {"message": "Scored by and assisted by is needed"}
+
+    try:
+        scored_by = Player.objects.get(id=match_event.scored_by_id)
+        assisted_by = Player.objects.get(id=match_event.assisted_by_id)
+    except Player.DoesNotExist:
+        return 422, {"message": "Invalid Scored by or assisted by"}
+
+    if team.id != match.stats.current_possession.id:
+        return 422, {"message": "Team does not match the team in current possession"}
+
+    # previous_score_events = MatchEvent.objects.filter(stats=match.stats, team=team, type=MatchEvent.EventType.SCORE).order_by("-time")
+
+    # if not previous_score_events:
+    #     team_that_started_on_offense = match.stats.initial_possession.id
+    #     started_on = MatchEvent.Mode.OFFENSE if team.id == team_that_started_on_offense else MatchEvent.Mode.DEFENSE
+    # else:
+    #     latest_score_event = previous_score_events[0]
+    #     started_on =
+
+    new_match_event = MatchEvent(
+        stats=match.stats,
+        team=team,
+        started_on=MatchEvent.Mode.OFFENSE,  # ignore this for now, not used
+        scored_by=scored_by,
+        assisted_by=assisted_by,
+        type=MatchEvent.EventType.SCORE,
+    )
+    new_match_event.save()
+
+    if match.team_1 is not None and match.team_2 is not None and team.id == match.team_1.id:
+        match.stats.score_team_1 += 1
+        match.stats.current_possession = match.team_2
+    if match.team_1 is not None and match.team_2 is not None and team.id == match.team_2.id:
+        match.stats.score_team_2 += 1
+        match.stats.current_possession = match.team_1
+
+    match.stats.save()
+
+    return 200, match.stats
+
+
+def handle_score_undo(
+    match: Match, score_event: MatchEvent
+) -> tuple[int, MatchStats | message_response]:
+    if match.team_1 is None or match.team_2 is None:
+        return 422, {"message": "Invalid teams"}
+
+    if score_event.team.id == match.team_1.id:
+        match.stats.score_team_1 -= 1
+        match.stats.current_possession = match.team_1
+
+    if score_event.team.id == match.team_2.id:
+        match.stats.score_team_2 -= 1
+        match.stats.current_possession = match.team_2
+
+    score_event.delete()
+    match.stats.save()
+
+    return 200, match.stats
+
+
+def handle_block(
+    match_event: MatchEventCreateSchema, match: Match, team: Team
+) -> tuple[int, MatchStats | dict[str, str]]:
+    if match_event.block_by_id is None:
+        return 422, {"message": "Block by is needed"}
+
+    try:
+        block_by = Player.objects.get(id=match_event.block_by_id)
+    except Player.DoesNotExist:
+        return 422, {"message": "Invalid Block by"}
+
+    if team.id == match.stats.current_possession.id:
```

Comment:
this is wrong? current possession is the team on O?

Reply:
so current possession is team on O. `team` here is the team of the player who got the block right ? So the check will fail if the team on O gets a block, which cant happen